Credentials/credentials_student.php

```diff
@@ -25,134 +32,94 @@
     echo '</div>';
 } else {
 
-    $gibbonPersonID = isset($_GET['gibbonPersonID'])? $_GET['gibbonPersonID'] : '';
-    $search = isset($_GET['search'])? $_GET['search'] : '';
-    $allStudents = isset($_GET['allStudents'])? $_GET['allStudents'] : '';
+    $gibbonPersonID = $_GET['gibbonPersonID'] ?? '';
+    $search = $_GET['search'] ?? '';
+    $allStudents = $_GET['allStudents'] ?? '';
 
-    if ($gibbonPersonID == false) { echo "<div class='error'>";
+    if ($gibbonPersonID == '') {
+        echo "<div class='error'>";
         echo __('You have not specified one or more required parameters.');
         echo '</div>';
     } else {
         try {
-            if ($allStudents != 'on') {
-                $data = array('gibbonSchoolYearID' => $_SESSION[$guid]['gibbonSchoolYearID'], 'gibbonPersonID' => $gibbonPersonID);
-                $sql = "SELECT * FROM gibbonPerson JOIN gibbonStudentEnrolment ON (gibbonPerson.gibbonPersonID=gibbonStudentEnrolment.gibbonPersonID) WHERE gibbonSchoolYearID=:gibbonSchoolYearID AND status='Full' AND (dateStart IS NULL OR dateStart<='".date('Y-m-d')."') AND (dateEnd IS NULL  OR dateEnd>='".date('Y-m-d')."') AND gibbonPerson.gibbonPersonID=:gibbonPersonID";
-            } else {
-                $data = array('gibbonPersonID' => $gibbonPersonID);
-                $sql = 'SELECT DISTINCT gibbonPerson.* FROM gibbonPerson LEFT JOIN gibbonStudentEnrolment ON (gibbonPerson.gibbonPersonID=gibbonStudentEnrolment.gibbonPersonID) WHERE gibbonPerson.gibbonPersonID=:gibbonPersonID';
-            }
-            $result = $connection2->prepare($sql);
-            $result->execute($data);
-        } catch (PDOException $e) {
-            echo "<div class='error'>".$e->getMessage().'</div>';
+            $gibbonSchoolYearID = $_SESSION[$guid]['gibbonSchoolYearID'];
+
+            $studentGateway = $container->get(CredentialsCredentialGateway::class);
+            $searchColumns = $studentGateway->getSearchableColumns();
+
+            $criteria = $studentGateway->newQueryCriteria()
+                            ->searchBy($searchColumns, $search)
+                            ->sortBy(['surname', 'preferredName'])
+                            ->filterBy('all', $allStudents ?? '')
+                            ->fromPOST();
+            $students = $studentGateway->queryStudentBySchoolYear($criteria, $gibbonSchoolYearID, $gibbonPersonID);
+        } catch (Exception $e) {
```

Comment:
Gateway classes now handle exceptions internally, so any try-catch blocks can be removed too :D